collectors/sophossiem/collector.js

```diff
@@ -0,0 +1,141 @@
+/* -----------------------------------------------------------------------------
+ * @copyright (C) 2020, Alert Logic, Inc
+ * @doc
+ *
+ * sophossiem class.
+ *
+ * @end
+ * -----------------------------------------------------------------------------
+ */
+'use strict';
+
+const moment = require('moment');
+const PawsCollector = require('@alertlogic/paws-collector').PawsCollector;
+const parse = require('@alertlogic/al-collector-js').Parse;
+const packageJson = require('./package.json');
+const utils = require("./utils");
+
+
+const typeIdPaths = [{ path: ['id'] }];
+
+const tsPaths = [{ path: ['created_at'] }];
+
+
+class SophossiemCollector extends PawsCollector {
+    constructor(context, creds) {
+        super(context, creds, packageJson.version);
+    }
+
+    pawsInitCollectionState(event, callback) {
+        const startTs = process.env.paws_collection_start_ts ?
+            process.env.paws_collection_start_ts :
+            moment().toISOString();
+
+        //From date must be within last 24 hours
+        let from_date_unix;
+        if (moment().diff(startTs, 'hours') < 24) {
+            from_date_unix = moment(startTs).unix();
+        }
+        else {
+            from_date_unix = moment().subtract(23, 'hours').unix();
+        }
+
+        const objectNames = JSON.parse(process.env.paws_collector_param_string_1);
+        const initialStates = objectNames.map(objectName => {
+            return {
+                objectName,
+                from_date: from_date_unix,
+                poll_interval_sec: 1
+            }
+        });
+
+        return callback(null, initialStates, 1);
+    }
+
+    pawsGetRegisterParameters(event, callback) {
+        const regValues = {
+            sophosSiemObjectNames: process.env.paws_collector_param_string_1
+        };
+        callback(null, regValues);
+    }
+
+    pawsGetLogs(state, callback) {
+        let collector = this;
+
+        const clientSecret = collector.secret;
+        if (!clientSecret) {
+            return callback("The Authorization token was not found!");
+        }
+
+        const x_api_key = collector.clientId;
+        if (!x_api_key) {
+            return callback("The x-api-key was not found!");
+        }
+
+        const headers = {
+            "x-api-key": x_api_key,
+            "Authorization": clientSecret
+        };
+
+        const APIHostName = collector.pawsDomainEndpoint;
+        if (!APIHostName) {
+            return callback("The Host Name was not found!");
+        }
+
+        console.info(`SIEM000001 Collecting data for ${state.objectName}`);
```

Comment:
I would prefer that we mark the time range of the query here as well. thats really helpful for determining if a collector is caught up or not.

Reply:
we need to decode that next_cursor value and extract timestamp value from that string.

Reply:
It would be good if we could. How feasable do you think that is?

Reply:
okay sure, I'll try

Reply:
If logs available and pagination true then next_cursor return this string 
V2_CURSOR|2020-07-31T00:00:00.000Z|2020-07-31T08:47:38.849Z|es:[1596154817562,"asdsadasdasdasdasdasdas"]:10
V2_CURSOR|2020-07-31T00:00:00.000Z|2020-07-31T08:47:38.849Z|es:[1596156008047,"asdsadasssssasdACACASCS"]:10


if logs available and pagination false then next_cursor return this string 
V2_CURSOR|2020-07-31T08:55:51.970Z

if logs not available and pagination false then next_cursor return this string 
0|2020-07-31T08:57:11.071Z

Reply:
Done